### ISSUE TITLE
Zoom into cursor position instead of the center of the image.

### ISSUE THREAD

**Issue by THEGOLDENPRO** [cloudy-org/roseate#17]
~~Required to be completed before we can complete and merge #12~~

- [x] Have image zoom into cursor
- [ ] If the image is smaller than the window switch to zooming the image up in the center of the window instead of the cursor pos

**Comment by THEGOLDENPRO** [comment]
This has been proven to be very very challenging in the codebase currently. 

Just a note to myself:
- [ ] Move all scaling and image sizing logic to ZoomPan.

**Comment by THEGOLDENPRO** [comment]
yeah it's not looking good, I can't figure this one out and the egui transformation examples are not working so this issue will postponed for now.
